src/App.tsx

```diff
@@ -16,77 +25,128 @@ const stationMap = new Map<string, string>([
+        const originStation = stationMap.get(originStationName);
+        const destinationStation = stationMap.get(destinationStationName);
+
+        if (!originStation || !destinationStation) {
+            return null;
+        }
+
+        return [
+            apiUrlMaker({
+                originStation,
+                destinationStation,
+                outboundDateTime: new Date(Date.now() + 10000).toISOString(),
+                numberOfChildren: '0',
+                numberOfAdults: '1',
+            }),
+            {
+                method: 'GET',
+                headers: {
+                    'X-API-KEY': `${process.env.REACT_APP_X_API_KEY}`,
+                },
+            },
+        ];
+    };
+
+    const fetchTrainData = async () => {
+        const requestOptions = getTrainRequestOptionsOrNull();
+
+        if (!requestOptions) {
+            return;
+        }
+
+        setIsLoading(true);
+        const response = await fetch(...requestOptions);
+        const json = await response.json();
+        setIsLoading(false);
+        setTrainData(json);
+    };
 
     React.useEffect(() => {
         setDisableSubmit(
-            !departureStation || !arrivalStation || departureStation===arrivalStation,
+            !originStationName
+            || !destinationStationName
+            || originStationName===destinationStationName,
         );
-    }, [departureStation, arrivalStation]);
+    }, [originStationName, destinationStationName]);
 
     const onSubmit = () => {
-        departureStation
-        && arrivalStation
-        && window.open(urlMaker(departureStation, arrivalStation));
+        originStationName
+        && destinationStationName
+        && fetchTrainData();
     };
 
     const getDisabledMessage = () => {
-        return <div className = "disable-message has-text-danger" >
+        return <div className = "disable-message" >
             {
-                (!departureStation || !arrivalStation)
+                (!originStationName || !destinationStationName)
             && <>Select two stations.</>
             || <>Departure station cannot be the same as arrival station.</>
             }
         </div>;
     };
 
     return <BrowserRouter>
-        <div className = "App">
-            <div className = "dropdown-menus-container">
-                <StationDropdown
-                    valueUpdateFunction = { setDepartureStation }
-                    label = 'Departure:'
-                    selectableStations = { stationNames }
-                    id = 'departure-station-selection'
-                />
-                <StationDropdown
-                    valueUpdateFunction = { setArrivalStation }
-                    label = 'Arrival:'
-                    selectableStations = { stationNames }
-                    id = 'arrival-station-selection'
-                />
-            </div>
+        <div className = "user-entry">
+            <div className = "App">
```

Comment:
is there a reason for user-entry and App being separate divs? Happy to hear the yes reason if you have thought this through :)

Reply:
No reason - removed it :)